h2o-py/tests/testdir_algos/automl/pyunit_automl_args.py

```diff
@@ -260,6 +261,24 @@ def test_automl_stops_after_max_runtime_secs():
     assert abs(end-start - max_runtime_secs) < cancel_tolerance_secs, end-start
 
 
+def test_no_model_takes_more_than_max_runtime_secs_per_model():
+    print("Check that individual model get interrupted after `max_runtime_secs_per_model`")
+    ds = import_dataset(seed=1, larger=True)
+    max_runtime_secs = 30
+    models_count = {}
+    for max_runtime_secs_per_model in [0, 3, max_runtime_secs]:
+        aml = H2OAutoML(project_name="py_aml_max_runtime_secs_per_model_{}".format(max_runtime_secs_per_model), seed=1,
+                        max_runtime_secs_per_model=max_runtime_secs_per_model,
+                        max_runtime_secs=max_runtime_secs)
+        aml.train(y=ds['target'], training_frame=ds['train'])
+        models_count[max_runtime_secs_per_model] = len(aml.leaderboard)
+        # print(aml.leaderboard)
+    # there may be one model difference as reproducibility is not perfectly guaranteed in time-bound runs
+    assert abs(models_count[0] - models_count[max_runtime_secs]) <= 1
```

Comment:
@seb-h2o it means that in `models_count[max_runtime_secs]` case we might not get a model in the leaderboard? Why then for the `models_count[3]` case we are expecting one model to be built? `models_count[max_runtime_secs]` has more time then `models_count[3]` and intuitively I would expect it to deliver more models.

Reply:
there's no such assumption.
Just testing here that when setting `max_runtime_secs_per_model` to 0 (ie. disabled) or to `max_runtime_secs` (full time), then we get same amount of models (with a tolerance of 1).
But when setting it to 3s, then we get much less models, than when it's disabled (=0).